crates/bevy_reflect/bevy_reflect_derive/src/field_attributes.rs

```diff
@@ -69,85 +71,150 @@ pub(crate) enum DefaultBehavior {
-                    Err(syn::Error::new(
-                        err.span(),
-                        format!("expected a string literal containing the name of a function, but found: {}", err.to_token_stream()),
-                    ))
+            }
+            Meta::Path(path) if path.is_ident(IGNORE_ALL_ATTR) => {
+                if args.ignore == ReflectIgnoreBehavior::None {
+                    args.ignore = ReflectIgnoreBehavior::IgnoreAlways;
+                    self.last_ignored = Some(path.span());
+                    Ok(())
+                } else {
+                    Err(syn::Error::new_spanned(path, format!("only one of ['{IGNORE_SERIALIZATION_ATTR}','{IGNORE_ALL_ATTR}'] is allowed")))
                 }
             }
-        }
-        Meta::NameValue(pair) => {
-            let path = &pair.path;
-            Err(syn::Error::new(
+            Meta::Path(path) if path.is_ident(DEFAULT_ATTR) => {
+                args.default = DefaultBehavior::Default;
+                Ok(())
+            }
+            Meta::Path(path) => Err(syn::Error::new(
                 path.span(),
                 format!("unknown attribute parameter: {}", path.to_token_stream()),
-            ))
+            )),
+            Meta::NameValue(pair) if pair.path.is_ident(DEFAULT_ATTR) => {
+                let lit = &pair.lit;
+                match lit {
+                    Lit::Str(lit_str) => {
+                        args.default = DefaultBehavior::Func(lit_str.parse()?);
+                        Ok(())
+                    }
+                    err => {
+                        Err(syn::Error::new(
+                            err.span(),
+                            format!("expected a string literal containing the name of a function, but found: {}", err.to_token_stream()),
+                        ))
+                    }
+                }
+            }
+            Meta::NameValue(pair) => {
+                let path = &pair.path;
+                Err(syn::Error::new(
+                    path.span(),
+                    format!("unknown attribute parameter: {}", path.to_token_stream()),
+                ))
+            }
+            Meta::List(list) if !list.path.is_ident(REFLECT_ATTRIBUTE_NAME) => {
+                Err(syn::Error::new(list.path.span(), "unexpected property"))
+            }
+            Meta::List(list) => {
+                for nested in &list.nested {
+                    if let NestedMeta::Meta(meta) = nested {
+                        self.parse_meta(args, meta)?;
+                    }
+                }
+                Ok(())
+            }
         }
-        Meta::List(list) if !list.path.is_ident(REFLECT_ATTRIBUTE_NAME) => {
-            Err(syn::Error::new(list.path.span(), "unexpected property"))
+    }
+
+    /// Verifies `#[reflect(ignore)]` attributes are always last in the type definition.
+    fn check_ignore_order(&self, args: &ReflectFieldAttr, errors: &mut Option<syn::Error>) {
```

Comment:
This is a logical fix to a real problem! Users deriving reflect like this already have control over their field orders, so asking them to change the order in these cases seems reasonable in _most_ cases. However, theres always the chance that field order matters for some other reason (ex: the type is also used for binary serialization: RPC, bincode, GPU types, etc).

Given that Reflect aims to be a "general purpose reflection library", I think breaking these cases will eventually be an issue, both in Bevy projects and elsewhere.

Reply:
Fair enough. Would it be better to put this behind a default feature? Or at least provide an opt-out attribute like `#[reflect(allow_ignore_order)]` (or some other better name lol)?

Reply:
If opting out breaks the scenarios we're trying to fix in this PR (FromReflect and serialization), and this PR breaks other scenarios (reflecting structs that needs specific field orders), I think thats an indicator that we need to rethink our implementation generally / come up with a solution that handles all of these scenarios correctly.

Specifically: is it possible for us to support cases where reflected field indices don't line up exactly with declared field indices. Or alternatively, can we adjust our reflection apis to account for "non-existent / skipped" fields (ensuring reflected field indices always match declared field indices). If either of these are possible, what are the costs?  Performance? Ergonomics? Do we need to mitigate those costs?

Reply:
> Specifically: is it possible for us to support cases where reflected field indices don't line up exactly with declared field indices. Or alternatively, can we adjust our reflection apis to account for "non-existent / skipped" fields (ensuring reflected field indices always match declared field indices).

I just put up an alternative PR (#7575) which is more in line with this. The only downside is that it requires fields with `#[reflect(skip_serializing)]` to provide some way of generating a default instance (either with a `Default` impl or with a custom function). As noted in the PR, though, I think it makes sense to add such a requirement.

Reply:
Brilliant!